### PR TITLE
SWATCH-3246: fix subscription configuration properties and add subscription sync component tests

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: SWATCH-3246

## Changes
The problem is that in clowdapp.yaml, we use SUBSCRIPTION_IGNORE_EXPIRED_OLDER_THAN=2M and SUBSCRIPTION_IGNORE_STARTING_LATER_THAN=2M.
In Spring, these properties standed for 2 months because it uses a custom duration parser, but in Quarkus, 2M stands for 2 minutes. 
The fix is to represent the duration in days: 60d. 

## Testing
Added four scenarios to extend our coverage for the subscription sync scenario: 
- subscriptions-sync-TC003 - Empty upstream org subscription list
- subscriptions-sync-TC004 - Upstream org subscription search failure
- subscriptions-sync-TC005 - Subscription missing from subscription search list
- subscriptions-sync-TC006 - Expired and replacement both returned from subscription search
- subscriptions-sync-TC007 - Subscription filtered because upstream start date is null
- subscriptions-sync-TC008 - Subscription filtered because upstream start date is too far in the future

## Release Notes

* **New Features**
  * Added date-range filtering capability for SKU capacity queries.

* **Tests**
  * Expanded subscription synchronization test coverage with six new scenarios covering reconciliation, search failures, subscription replacements, and edge cases.

### CODE REVIEW

File: swatch-contracts/TEST_PLAN.md [hunk 1]
```diff
@@ -904,9 +904,9 @@ This section verifies the automatic contract termination behavior when contracts
 - **Note:** This endpoint **SUPPORTS multiple subscriptions** via JSON array
 
 **subscriptions-sync-TC001 - Sync all subscriptions for enabled orgs**  
-- **Description**: Verify POST /rpc/subscriptions/sync enqueues org subscriptions.  
+- **Description**: Verify PUT /rpc/subscriptions/sync enqueues org subscriptions.  
 - **Setup**: Configure sync-enabled orgs.  
-- **Action:** POST `/api/swatch-contracts/internal/rpc/subscriptions/sync`.  
+- **Action:** PUT `/api/swatch-contracts/internal/rpc/subscriptions/sync`.
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply swatch-contracts/TEST_PLAN.md]: fixed

File: swatch-contracts/ct/java/tests/SubscriptionsSyncComponentTest.java [hunk 2]
```diff
@@ -60,10 +86,261 @@ void shouldSyncUmbSubscriptionMessage() {
...
+                + "–"
+                + inclusiveTo);
+      } else {
+        assertEquals(
+            expectedSockets,
+            actual,
+            0.01,
+            "Capacity on " + day + " should report expected socket capacity");
+      }
+    }
```
Comment (Mantzounis) [inline swatch-contracts/ct/java/tests/SubscriptionsSyncComponentTest.java:329]: is the a reason for calling the GET inside the for loop here and in the above methods? If not could we get the response once and then filter inside the for loop?

**Review by Mantzounis — CHANGES_REQUESTED** [review]
question: I noticed the additions in domain are not called by any tests, what is the reason for adding them

File: swatch-contracts/ct/java/tests/SubscriptionsSyncComponentTest.java (hunk 2 above)
Comment (Sgitario) [reply swatch-contracts/ct/java/tests/SubscriptionsSyncComponentTest.java:329]: I've refactored this method, let me know what you think.

**Comment by Sgitario** [comment]
> question: I noticed the additions in domain are not called by any tests, what is the reason for adding them

I had thought to keep those methods because it was very useful during the investigation, but I'm fine removing them, we can always add it back when it's really used.

**Comment by Mantzounis** [comment]
@Sgitario I'm fine with keeping them there if they are useful for debugging but we should document the decision in a comment to remember its intentional

**Comment by Sgitario** [comment]
> @Sgitario I'm fine with keeping them there if they are useful for debugging but we should document the decision in a comment to remember its intentional

no problem, just removed it. we can always add it back, it's not super complicated.

**Review by Mantzounis — APPROVED** [review]
thanks for the updates! lgtm.
